desc/nestor.py

```diff
@@ -0,0 +1,1211 @@
+"""Neumann Solver for Toroidal Systems.
+
+Original python implementation by Jonathan Schilling (jonathan.schilling@ipp.mpg.de)
+Rewritten to use JAX by DESC team.
+"""
+
+import numpy as np
+
+from desc.backend import fori_loop, jnp, put
+from desc.grid import LinearGrid
+from desc.io import IOAble
+from desc.transform import Transform
+from desc.utils import Index, setdefault
+
+
+def copy_vector_periods(vec, zetas):
+    """Copies a vector into each field period by rotation.
+
+    Parameters
+    ----------
+    vec : ndarray, shape(3,...)
+        vector(s) to rotate
+    zetas : ndarray
+        angles to rotate by (eg start of each field period)
+
+    Returns
+    -------
+    vec : ndarray, shape(3,...,nzeta)
+        vector(s) repeated and rotated by angle zeta
+    """
+    if vec.shape[0] == 3:
+        x, y, z = vec
+    else:
+        x, y = vec
+    shp = x.shape
+    xx = x.reshape((*shp, 1)) * jnp.cos(zetas) - y.reshape((*shp, 1)) * jnp.sin(zetas)
+    yy = y.reshape((*shp, 1)) * jnp.cos(zetas) + x.reshape((*shp, 1)) * jnp.sin(zetas)
+    if vec.shape[0] == 3:
+        zz = jnp.broadcast_to(z.reshape((*shp, 1)), (*shp, zetas.size))
+        return jnp.array((xx, yy, zz))
+    return jnp.array((xx, yy))
+
+
+def eval_surface_geometry(
+    R_lmn, Z_lmn, Rb_transform, Zb_transform, ntheta, nzeta, NFP, sym
+):
+    """Evaluate coordinates and derivatives on the surface.
```

Comment:
These are mostly from Jonathan. We could replace most of these with our compute stuff but I'd prefer minimize the changes to all the nestor stuff

Reply:
OK this might be controversial, but I think it's worth a discussion: What if we don't include the NESTOR stuff in this PR? 

My reasoning:

1. The code is messy. 
2. It's worse than the BIEST implementation and we don't recommend using it. 
3. We don't have any tutorials, example, or tests for the NESTOR code. 

We are adding this whole new file of duplicate code that does not conform to the API of the rest of the DESC code. Moreover, _we don't even use this code_! It sounds like there is no use case for the NESTOR stuff in the near future, but if we ever decide that we do need it we can easily add it back.

Reply:
Since it is used in the paper I agree it is worth keeping in the code repo. I am fine with it not being in the API documentation maybe with a note in the docstring saying it is not meant for public use or something